### ISSUE TITLE
Homework #4 - numpy

### ISSUE THREAD

**Issue by cmurray3** [IE-555/spring2023#17]
Homework 4 is now available:
- https://github.com/IE-555/spring2023/blob/main/Homework/HW_4_numpy/HW_4_numpy.pdf
If you have any questions, please post them below.

Please click this link to create your GitHub repository (to submit your assignment):
https://classroom.github.com/a/IZOSQTcf

**Comment by arvindUB** [comment]
how do we deal with nan values here? If a student misses an assignment they would get a 0. So should we replace all nan with 0s here?

What about the student who dropped the class (ID: 1234)? Should they get 0 in all missed exams and hw?

**Comment by cmurray3** [comment]
> how do we deal with nan values here? If a student misses an assignment they would get a 0. So should we replace all nan with 0s here?
> 
> What about the student who dropped the class (ID: 1234)? Should they get 0 in all missed exams and hw?

Instructions about how to handle missing data are covered in the sample data file:
- https://github.com/IE-555/spring2023/blob/main/Homework/HW_4_numpy/grades_example.csv

**Comment by atharvavt** [comment]
1. In sub section C & D. You have asked to find students who has more than 90 score in HW1 and Hw3. But Data which you have provided in Hw1 has 10 and Hw3 has 50 marks weightage. Can you please elaborate on the required output?

2. As example data provided by you has string data including first 2 rows and last 9 rows. So, while writing function do we need to remove those values or when you are going to test our code you will provide non-string values only? 
(Do we need to remove or handle sting data using NumPy)?

**Comment by cmurray3** [comment]
> * In sub section C & D. You have asked to find students who has more than 90 score in HW1 and Hw3. But Data which you have provided in Hw1 has 10 and Hw3 has 50 marks weightage. Can you please elaborate on the required output?

Here, "90" refers to the percentage, which you can calculate from the marks earned and the marks available for each homework.

> * As example data provided by you has string data including first 2 rows and last 9 rows. So, while writing function do we need to remove those values or when you are going to test our code you will provide non-string values only?
>   (Do we need to remove or handle sting data using NumPy)?

Please read `grades_example.csv`, which describes the structure of the input file(s) on which your code will be tested, carefully.

**Comment by cmurray3** [comment]
> 1. I’ve been stuck with the assignment at question-e, which is "Find the weighted scores of each student". Do we need to generalize this while coding it or can we code in such a way that it works for this particular dataset? Because the columns might change in a new dataset.

Yes, the columns will absolutely change in the new test dataset.  You will need to generalize the code to work with input files that are consistent with the file structure defined in the `grades_example.csv` file.

> 2. And where to use the input numExams in this code?
The value of `numExams` is necessary to determine which columns of the input data correspond to exams versus homework.  

> A clarification on this is much needed. Sorry for the late-minute doubts.

**Comment by cmurray3** [comment]
In case I forgot to mention this before, the deadline to submit your code (via the link in the first message on this page) is 11:59pm tonight.
